glib-macros/src/lib.rs

```diff
@@ -310,3 +313,76 @@ pub fn gflags(attr: TokenStream, item: TokenStream) -> TokenStream {
     let gen = gflags_attribute::impl_gflags(&input, &gtype_name);
     gen.into()
 }
+
+/// Macro for deriving implementations of [`glib::Downgrade`] and
+/// [`glib::Upgrade`] traits and a weak type.
+///
+/// # Examples
+///
+/// ## New Type Idiom
+///
+/// ```rust,ignore
+/// #[derive(Downgrade)]
+/// pub struct FancyLabel(gtk::Label);
+///
+/// impl FancyLabel {
+///     pub fn new(label: &str) -> Self {
+///         Self(gtk::LabelBuilder::new().label(label).build())
+///     }
+///
+///     pub fn flip(&self) {
+///         self.0.set_angle(180.0 - self.0.get_angle());
+///     }
+/// }
+///
+/// let fancy_label = FancyLabel::new("Look at me!");
+/// let button = gtk::ButtonBuilder::new().label("Click me!").build();
+/// button.connect_clicked(clone!(@weak fancy_label => move || fancy_label.flip()));
+/// ```
+///
+/// ## Generic New Type
+///
+/// ```rust,ignore
+/// #[derive(Downgrade)]
+/// pub struct TypedEntry<T>(gtk::Entry, std::marker::PhantomData<T>);
+///
+/// impl<T: ToString + FromStr> for TypedEntry<T> {
+///     // ...
+/// }
+/// ```
+///
+/// ## Structures and Enums
+///
+/// ```rust,ignore
+/// #[derive(Clone, Downgrade)]
```

Comment:
Isn't it supposed to be `#[derive(glib::Downgrade)]`?

Reply:
Done